File: src/main/java/com/carolin_violet/travel_system/bean/Article.java
package com.carolin_violet.travel_system.bean;

import com.baomidou.mybatisplus.annotation.*;

import java.util.Date;

import java.io.Serializable;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

/**
 * <p>
 * 
 * </p>
 *
 * @author carolin-violet
 * @since 2022-08-21
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
public class Article implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 介绍文章id
     */
    @TableId(value = "id", type = IdType.ID_WORKER_STR)
    private String id;

    /**
     * 介绍文章标题
     */
    private String title;

    /**
     * 介绍文章详细内容(html内容)
     */
    private String contentHtml;

    /**
     * 介绍文章详细内容(普通文本内容)
     */
    private String contentText;

    /**
     * 介绍文章标志(如美食、景点、酒店、公告等)
     */
    private String mark;

    /**
     * 对应的(美食、景点、酒店、公告的id)
     */
    private String markId;

    /**
     * 逻辑删除
     */
    @TableLogic
    @TableField(fill = FieldFill.INSERT)
    private Integer isDeleted;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private Date createTime;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Date modifyTime;


}

File: server/model/article.js
const Sequelize = require('sequelize')
const MySequelize = require('../config/mysqlConfig')

const articleModel = MySequelize.define('article', {
  id: {
    type: Sequelize.STRING(19),
    field:  'id',
    primaryKey: true
  },
  title: {
    type: Sequelize.STRING(25),
    field:  'title',
    allowNull: false
  },
  content_html: {
    type: Sequelize.TEXT('long'),
    field:  'content_html',
    allowNull: false
  },
  content_text: {
    type: Sequelize.TEXT('long'),
    field:  'content_text',
    allowNull: false
  },
  mark: {
    type: Sequelize.STRING(10),
    field:  'mark',
    allowNull: false
  },
  mark_id: {
    type: Sequelize.STRING(19),
    field:  'mark_id',
    allowNull: false
  },
  is_deleted: {
    type: Sequelize.TINYINT,
    field:  'is_deleted',
    allowNull: false
  },
  create_time: {
    type: Sequelize.DATE,
    field:  'create_time',
    allowNull: false
  },
  modify_time: {
    type: Sequelize.DATE,
    field:  'modify_time',
    allowNull: false
  },
})

module.exports = articleModel